pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -117,11 +125,80 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 	err = c.Watch(&source.Kind{Type: &mapi.Machine{
 		ObjectMeta: meta.ObjectMeta{Namespace: "openshift-machine-api"},
-	}}, &handler.EnqueueRequestForObject{}, predicateFilter)
+	}}, &handler.EnqueueRequestForObject{}, machinePredicate)
 	if err != nil {
 		return errors.Wrap(err, "could not create watch on Machine objects")
 	}
 
+	err = c.Watch(&source.Kind{Type: &core.Node{
+		ObjectMeta: meta.ObjectMeta{Namespace: ""},
+	}}, &handler.EnqueueRequestsFromMapFunc{ToRequests: newNodeToMachineMapper(mgr.GetClient())}, predicate.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent) bool {
+			if createEvent.Meta.GetAnnotations()[nodeconfig.VersionAnnotation] != version.Get() {
```

Comment:
nit: This will filter out all linux nodes by default, but it might be good to be more verbose, and return if the node is not a Windows node

applies to all predicate funcs

Reply:
We are checking for label [here](https://github.com/openshift/windows-machine-config-operator/pull/101/files#diff-d5d428eaf9d5786d351222f5184d99d9R177)